plugins/ctf/fs-sink/write.c

```diff
@@ -394,12 +357,7 @@ struct fs_writer *insert_new_writer(
 	} else {
 		ret = bt_ctf_trace_add_is_static_listener(trace,
 				trace_is_static_listener, fs_writer);
-		if (ret < 0) {
-			fprintf(writer_component->err,
-					"[error] %s in %s:%d\n", __func__, __FILE__,
-					__LINE__);
-			goto error;
-		}
+		assert(ret >= 0);
```

Comment:
`bt_ctf_trace_add_is_static_listener()` can fail if the trace is already static. Are you sure?